File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int m = *max_element(candies.begin(), candies.end());
        vector<bool> ans;
        for(auto x:candies){
            ans.push_back((x+extraCandies >= m));
        }
        return ans;
    }
};

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.rb
# @param {Integer[]} candies
# @param {Integer} extra_candies
# @return {Boolean[]}
def kids_with_candies(candies, extra_candies)
    val = candies.max
    n = candies.size
    res = []
    (0..n-1).each do |i|
        if(candies[i]+extra_candies >=val)
            res << true
        else
            res << false
        end
    end
    res
end